test/tokens/ZcToken.sol

```diff
@@ -33,11 +33,13 @@ contract ZcToken is Erc2612, IZcToken {
       return true;
   }
 
+  /// @dev Muturity must not have been reached else we revert here.
   /// @param t Address recieving the minted amount
   /// @param a The amount to mint
   function mint(address t, uint256 a) external onlyAdmin(admin) override returns(bool) {
-      _mint(t, a);
-      return true;
+    require(block.timestamp < maturity, 'maturity reached');
```

Comment:
Use a custom error here instead and revert to save gas.

Reply:
not introducing custom errors in V2.